src/semaphore.ts

```diff
@@ -15,10 +15,6 @@ export class Semaphore {
     return this._permits;
   }
 
-  get queueLength() {
-    return this._queue.length;
-  }
-
   public async acquire(): Promise<void> {
     if (this._permits > 0) {
       this._permits -= 1;
```

Comment:
### Code Review

#### Pros
- Removing `queueLength` property might streamline the Semaphore class interface if the queue length is considered an internal detail.

#### Cons
- Removal of `queueLength` could be considered a breaking change if external consumers depend on this property to make decisions. If so, their code will break after the update, which suggests that a major version bump should be considered if applying this change.
- The purpose and context of the patch are not provided. It is important to understand why the `queueLength` is being removed in order to properly assess the impact of such change.

#### Security Risks
- There are no obvious security risks associated with removing a getter for the internal `_queue` length within the provided code.

#### Suggestions for Improvement
- Ensure there is proper communication and versioning strategy to handle the potentially breaking change.
- Add tests or ensure existing tests cover cases where `queueLength` removal might affect semaphore functionality.

#### Overall Rating
Without additional context, it cannot be fully determined whether removing the `queueLength` is beneficial; thus, it is assumed neutral from a code health standpoint. Assuming no other issues are present in unshown parts of the patch, and assuming the removal does not negatively affect external dependencies, this would reflect a standard iterative code change.

Quality Rate: 7/10